upup/pkg/fi/nodeup/protokube_flags.go

```diff
@@ -29,5 +29,6 @@ type ProtokubeFlags struct {
 
 	DNSInternalSuffix *string `json:"dnsInternalSuffix,omitempty" flag:"dns-internal-suffix"`
 	Cloud             *string `json:"cloud,omitempty" flag:"cloud"`
-	ClusterId         *string `json:"cluster-id,omitempty" flag:"cluster-id"`
+	// ClusterId flag is required only for vSphere cloud type, to pass cluster id information to protokube. AWS and GCE workflows ignore this flag.
```

Comment:
How does AWS and GCE get the clusterId now for protokube? Could you explain why the same approach would not work for vSphere?